lib/cocoapods/target/pod_target.rb

```diff
@@ -818,33 +911,54 @@ def version
     #        whether to include header search paths for private headers of this
     #        target
     #
+    # @param [String] configuration
+    #        the configuration to return header search paths for or `nil` for all configurations.
+    #
     # @return [Array<String>] The set of header search paths this target uses.
     #
-    def header_search_paths(include_dependent_targets_for_test_spec: nil, include_dependent_targets_for_app_spec: nil, include_private_headers: true)
+    def header_search_paths(include_dependent_targets_for_test_spec: nil, include_dependent_targets_for_app_spec: nil,
+                            include_private_headers: true, configuration: nil)
       header_search_paths = []
       header_search_paths.concat(build_headers.search_paths(platform, nil, false)) if include_private_headers
       header_search_paths.concat(sandbox.public_headers.search_paths(platform, pod_name, uses_modular_headers?))
-      dependent_targets = recursive_dependent_targets
-      dependent_targets += recursive_test_dependent_targets(include_dependent_targets_for_test_spec) if include_dependent_targets_for_test_spec
-      dependent_targets += recursive_app_dependent_targets(include_dependent_targets_for_app_spec) if include_dependent_targets_for_app_spec
+      dependent_targets = recursive_dependent_targets(:configuration => configuration)
+      if include_dependent_targets_for_test_spec
+        dependent_targets += recursive_test_dependent_targets(include_dependent_targets_for_test_spec, :configuration => configuration)
+      end
+      if include_dependent_targets_for_app_spec
+        dependent_targets += recursive_app_dependent_targets(include_dependent_targets_for_app_spec, :configuration => configuration)
+      end
       dependent_targets.uniq.each do |dependent_target|
         header_search_paths.concat(sandbox.public_headers.search_paths(platform, dependent_target.pod_name, defines_module? && dependent_target.uses_modular_headers?(false)))
       end
       header_search_paths.uniq
     end
 
     # @param  [Specification] spec
-    #
+    # @param [String] configuration
```

Comment:
nit: alignment is off with the entries above and below

Reply:
fixing